index.js

```diff
@@ -29,27 +27,30 @@ class Registrar extends Emitter {
   /**
    * Add a registration for a user identified by a sip address-of-record
    * @param {String} aor - a sip address-of-record for a user (e.g. daveh@drachtio.org)
-   * @param {String} contact - the sip address where this user can be reached
-   * @param {String} sbcAddress - the sip uri address of the sbc that manages the connection to this user
-   * @param {String} protocol - the transport protocol used between the sbc and the user
+   * @param {String} obj.contact - the sip address where this user can be reached
+   * @param {String} obj.sbcAddress - the sip uri address of the sbc that manages the connection to this user
+   * @param {String} obj.protocol - the transport protocol used between the sbc and the user
    * @param {String} expires - number of seconds the registration for this user is active
    * @returns {Boolean} true if the registration was successfully added
    */
   async add(aor, obj, expires) {
     debug(`Registrar#add ${aor} from ${JSON.stringify(obj)} for ${expires}`);
     const key = makeUserKey(aor);
     try {
-      const now = Date.now();
-      obj.expiryTime = now + (expires * 1000);
+      //cleanup expired entries
```

Comment:
why are we doing this every time a registration is added?  In a very large system, couldn't we be adding tens of registrations per second?

Reply:
Good point, its only when doing a user count we should clean-up expired entries before the count. Will amend.

Reply:
These sorted set entries dont have an expiry, right?  So they do need to be manually cleaned up at some regular interval?  What if no one ever calls for a user count?  Should have some interval where we clean them up as well?